### ISSUE TITLE
DynamicParamValueProducer non-serializable?

### ISSUE THREAD

**Issue by NagyGa1** [omnifaces/omnifaces#140]
Oracle Weblogic 12.1.3.0 Update 2 validates beans at startup and throws this:

weblogic.management.DeploymentException: org.jboss.weld.exceptions.UnserializableDependencyException: WELD-001413 The bean Session bean [class com.cpox.view.common.ExceptionReportView with qualifiers [@Default @Any @Named]; local interfaces are [ExceptionReportView] declares passivating scope but has non-serializable dependency org.omnifaces.cdi.param.DynamicParamValueProducer@1430591d:org.jboss.weld.exceptions.UnserializableDependencyException:WELD-001413 The bean Session bean [class com.cpox.view.common.ExceptionReportView with qualifiers [@Default @Any @Named]; local interfaces are [ExceptionReportView] declares passivating scope but has non-serializable dependency org.omnifaces.cdi.param.DynamicParamValueProducer@1430591d

```java
@Named
@Stateful
@RequestScoped
public class ExceptionReportView implements Serializable {

	@Inject
	@Param
	private Long id;
```

Looking at the source, `DynamicParamValueProducer` implements `Serializable`, and its inner class `DefaultParamAnnotationLiteral` as well as it extends `AnnotationLiteral`, which is serializable as well.

So I don't really get what the problem is.

**Comment by arjantijms** [comment]
Haven't tested on 12.1.3 update 2 yet, but I'll have a look soon. Thanks for the report!

**Comment by NagyGa1** [comment]
That's a fast reply, thank you! :)

**Comment by NagyGa1** [comment]
Fiddled with it a few hours, cannot figure.

Weblogic uses Weld 2.1.0 (with probably some Oracle patches?) while Wildfly 8.2.0 uses 2.2.6.Final and GlassFish 4.1 uses 2.2.2.Final.

The thing works on Wildfly 8.2.0 and GlassFish 4.1, but does not on Weblogic. Might be a bug in Weblogic or in Weld 2.1. Checked Weld bugs closed or open, can't find anything.

Looked at the weld Validator code, implementing PassivationCapable on DynamicParamValueProducer should be enought. I have implemented it on DefaultParamAnnotationLiteral, made it Serializable, marked all fields transient, nothing helped.

Tried Omnifaces 1.6, 1.8.1, 1.8.2, 1.8.3, 2.0, there were various issues with the @Param injection in Wls 2.1.3.0 Update 2.

Workaround (long term) or me is to use ParamValue<>:
```java
	@Inject
	@Param
	private ParamValue<Long> id;
```
